miden-tx/tests/integration/scripts/p2id.rs

```diff
@@ -296,11 +292,15 @@ fn test_create_consume_multiple_notes() {
     assert_eq!(executed_transaction.output_notes().num_notes(), 2);
 
     account.apply_delta(executed_transaction.account_delta()).unwrap();
+
     for asset in account.vault().assets() {
-        if u64::from(asset.faucet_id()) == input_note_faucet_id {
-            assert!(asset.unwrap_fungible().amount() == 111);
-        } else if asset.faucet_id() == FungibleAsset::mock_issuer() {
-            assert!(asset.unwrap_fungible().amount() == 5);
+        // Avoid using Asset::faucet_id since it only returns the prefix.
+        if let Asset::Fungible(fungible) = asset {
```

Comment:
This should probably panic if the assets are not there/are not the same type. Also (not directly related to your changes) there should probably be an assertion on amount of assets since if for some reason the account ends up with no assets at this point, this test would pass anyway.